fs/sshfs/sshfs.py

```diff
@@ -155,6 +156,29 @@ def listdir(self, path):  # noqa: D102
         with convert_sshfs_errors('listdir', path):
             return self._sftp.listdir(_path)
 
+    def scandir(self, path, namespaces=None, page=None):  # noqa: D102
+        self.check()
+        _path = self.validatepath(path)
+        _namespaces = namespaces or ()
+        start, stop = page or (None, None)
+        try:
+            with convert_sshfs_errors('scandir', path, directory=True):
+                stat_iter = self._sftp.listdir_iter(_path)
```

Comment:
Just use `listdir_attr` here, this way you won't have to collect the results with `list` later!

Reply:
Done.  I was trying to preserve the responsiveness benefits of listdir_iter while paging, but maybe nobody uses paging with sshfs anyway, since we don't have a fast way to skip to the start.